File: app/src/main/java/pradyumna/simhansapp/prabandham/Prabandham.java
package pradyumna.simhansapp.prabandham;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;
import android.widget.Toolbar;

import java.util.ArrayList;

import pradyumna.simhansapp.R;
import pradyumna.simhansapp.adaptersFolders.RvAdapter;
import pradyumna.simhansapp.adaptersFolders.RvClickHandler;
import pradyumna.simhansapp.learn_veda.Veda_Player;
import pradyumna.simhansapp.viewModel.PrabandhamFolderViewModel;

public class Prabandham extends AppCompatActivity implements RvClickHandler {
    RecyclerView recyclerView;
    ArrayList<String> items;
    PrabandhamFolderViewModel mPrabandhamDataViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        //Status bar Color
        if (Build.VERSION.SDK_INT >= 21) {
            Window window = this.getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            window.setStatusBarColor(this.getResources().getColor(R.color.orangeMain));
        }

        //Action Bar Object
        ActionBar actionBar=getSupportActionBar();

        // Define ColorDrawable object and parse color
        ColorDrawable colorDrawable
                = new ColorDrawable(Color.parseColor("#F1D548"));

        // Set BackgroundDrawable
        actionBar.setBackgroundDrawable(colorDrawable);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_prabandham);

        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setDisplayHomeAsUpEnabled(true);

        recyclerView=findViewById(R.id.LprecyclerView);
        mPrabandhamDataViewModel= ViewModelProviders.of(this).get(PrabandhamFolderViewModel.class);

        RvAdapter adapter= new RvAdapter(this);
        recyclerView.setAdapter(adapter);
        LinearLayoutManager layoutManager=new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        if(mPrabandhamDataViewModel.getAllFolderName()!=null){
            mPrabandhamDataViewModel.getAllFolderName().observe(this, new Observer<ArrayList<String>>() {
                @Override
                public void onChanged(ArrayList<String> strings) {
                    if(!strings.isEmpty()){
                        items=strings;
                        adapter.setItems(strings);
                    }
                    else{
                        Log.d("TAG", "OnNoData");
                    }
                }
            });
        }
    }

    @Override
    public void onItemClick(int position) {
        Intent intent=new Intent(Prabandham.this, Prabandham_Player.class);
        intent.putExtra("Name",items.get(position));
        startActivity(intent);
    }
}

File: app/src/main/java/pradyumna/simhansapp/prabandham/Prabandham.kt
package pradyumna.simhansapp.prabandham

import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import pradyumna.simhansapp.R
import pradyumna.simhansapp.adaptersFolders.RvAdapter
import pradyumna.simhansapp.adaptersFolders.RvClickHandler
import pradyumna.simhansapp.viewModel.PrabandhamFolderViewModel
import java.util.*

class Prabandham : AppCompatActivity(), RvClickHandler {
    //Variables
    private var adapter = RvAdapter(this)
    private lateinit var recyclerView: RecyclerView
    lateinit var items: ArrayList<String?>
    lateinit var mPrabandhamDataViewModel: PrabandhamFolderViewModel
    override fun onCreate(savedInstanceState: Bundle?) {

        //Status bar Color
        if (Build.VERSION.SDK_INT >= 21) {
            val window = this.window
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            window.statusBarColor = this.resources.getColor(R.color.orangeMain)
        }

        //Action Bar Object
        val actionBar = supportActionBar

        // Define ColorDrawable object and parse color
        val colorDrawable = ColorDrawable(Color.parseColor("#F1D548"))

        // Set BackgroundDrawable
        actionBar!!.setBackgroundDrawable(colorDrawable)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_prabandham)
        actionBar.setDisplayHomeAsUpEnabled(true)
        actionBar.setDisplayHomeAsUpEnabled(true)

        //Set adapter
        recyclerView = findViewById(R.id.LprecyclerView)
        mPrabandhamDataViewModel = ViewModelProvider(this).get(PrabandhamFolderViewModel::class.java)
        recyclerView.adapter = adapter
        val layoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = layoutManager

        //Observing Live Data and providing to adapter
        if (mPrabandhamDataViewModel.allFolderName != null) {
            mPrabandhamDataViewModel.allFolderName.observe(this, { strings ->
                if (!strings.isEmpty()) {
                    strings.sort()
                    items = strings
                    adapter.submitList(strings)
                } else {
                    Log.d("TAG", "OnNoData")
                }
            })
        }
        //Search option filter data
        mPrabandhamDataViewModel.queryLiveData.observe(this, { s ->
            if (s != null) {
                val toList: List<String> = mPrabandhamDataViewModel.allFolderName.value?.filter { it.toLowerCase(Locale.ROOT).contains(s.toLowerCase(Locale.ROOT)) }?.toList()
                        ?: return@observe
                adapter.submitList(toList)
            }
        })
    }

    override fun onItemClick(position: Int) {
        val intent = Intent(this@Prabandham, Prabandham_Player::class.java)
        intent.putExtra("Name", items[position])
        startActivity(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {

        //Initialize Menu inflater
        val menuInflater: MenuInflater = menuInflater
        //Inflate Menu
        menuInflater.inflate(R.menu.menu_search, menu)
        //Initialize Menu Items
        val menuItem: MenuItem = menu?.findItem(R.id.search_view)!!

        //SearchView Variable
        val searchView: androidx.appcompat.widget.SearchView = menuItem.actionView as androidx.appcompat.widget.SearchView
        //Initialize Search View
        searchView.setOnQueryTextListener(object : androidx.appcompat.widget.SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                mPrabandhamDataViewModel.queryLiveData.postValue(query)
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                mPrabandhamDataViewModel.queryLiveData.postValue(newText)
                return true
            }
        })
        return super.onCreateOptionsMenu(menu)
    }
}